core/src/test/scala/integration/kafka/server/MultipleListenersWithSameSecurityProtocolTest.scala

```diff
@@ -113,7 +102,31 @@ class MultipleListenersWithSameSecurityProtocolTest extends ZooKeeperTestHarness
     * with acks=-1 to ensure that replication is also working.
     */
   @Test
-  def testProduceConsume(): Unit = {
+  def testMultipleListenersWithSameSecurityProtocol(): Unit = {
+    val plainSaslProperties = kafkaClientSaslProperties("PLAIN", dynamicJaasConfig = true)
+
+    val serverConfig = servers.head.config
+    serverConfig.listeners.foreach { endPoint =>
+      val listenerName = endPoint.listenerName
+
+      TestUtils.createTopic(zkUtils, listenerName.value, 2, 2, servers)
+
+      val trustStoreFile =
+        if (endPoint.securityProtocol == SecurityProtocol.SASL_SSL) Some(this.trustStoreFile)
+        else None
+      val SaslProperties =
+        if (listenerName.value == "SECURE_EXTERNAL" || listenerName.value == "SECURE_INTERNAL")
+          Some(plainSaslProperties)
```

Comment:
I think this is the only bit of code that needs to be different in the two tests. A common method that just sets saslProperties to `plainSaslProperties` for the first test and to `gssapiSaslProperties` or `plainSaslProperties` depending on listener name for the second test will reduce duplication and keep the two tests consistent.